_blog/2023-12-20-loco-rust-rails.mdx

```diff
@@ -0,0 +1,219 @@
+impl super::_entities::users::Model {
+    // .. your own methods
+}
+```
+
+Now we can use it in a handler function by loading the item from the database - then we can do whatever we need to with the data:
+```rust
+async fn load_item(ctx: &AppContext, id: i32) -> Result<Model> {
+    let item = Entity::find_by_id(id).one(&ctx.db).await?;
+    item.ok_or_else(|| Error::NotFound)
+}
+
+pub async fn update(
+    Path(id): Path<i32>,
+    State(ctx): State<AppContext>,
+    Json(params): Json<Params>,
+) -> Result<Json<Model>> {
+    // use sea_orm to load an item based on the id
+    let item = load_item(&ctx, id).await?;
+    
+    // turn the item into an ActiveModel that we can then use
+    let mut item = item.into_active_model();
+    
+    // update the parameters of the current item with the new properties
+    params.update(&mut item);
+    
+    // feed the new item back into the database
+    let item = item.update(&ctx.db).await?;
+    
+    // return the updated item
+    format::json(item)
+}
+```
+
+However - that isn't all that Loco.rs has to offer. We can also use the `loco_rs` Validator struct to be able to verify a new model before needing to do anything with it! A use case for this, for example, might be if we needed to check if an email is a valid email. You can check this out below:
+```rust
+[derive(Debug, Validate, Deserialize)]
+pub struct ModelValidator {
+    #[validate(length(min = 2, message = "Name must be at least 2 characters long."))]
+    pub name: String,
+    #[validate(custom = "validation::is_valid_email")]
+    pub email: String,
+}
+
+impl From<&ActiveModel> for ModelValidator {
+    fn from(value: &ActiveModel) -> Self {
+        Self {
+            name: value.name.as_ref().to_string(),
+            email: value.email.as_ref().to_string(),
+        }
+    }
+}
+```
+
+## Job Processing
+Like with everything else in Loco, you can also generate workers and tasks via the CLI. Running `cargo run generate task` or `cargo run generate worker` will let you generate a task or a worker at will.
+
+Under the hood, Loco uses `sidekiq-rs` to do job processing - which is a Rust re-implementation of its Ruby counterpart, `sidekiq.rb`. Once the worker is generated, you want to go to the `workers` folder and check out the file you made - it will have a struct for the worker itself, a struct that holds the arguments that the worker will take, an implementation of the `AppWorker` trait for the worker and then an async trait implementation that lets the worker do something. 
+
+You can then run it like so:
+```rust
+ReportWorkerWorker::perform
+    (
+        &boot.app_context, 
+        ReportWorkerWorkerArgs {}
+    )
+.await
+.unwrap();
+```
+
+As you can see, we don't need to initialise the struct to use it - we can call the method from the struct directly and it will work assuming the arguments are valid - as you can see here, no arguments are required since the args struct does not have any fields.
+
+## Deployment
```

Comment:
you might want to paste the actual command to generate a shuttle deployment so people can copy paste it :)